Student/Student/Assignments/AssignmentList/AssignmentListViewController.swift

```diff
@@ -19,49 +19,294 @@
+        //  setup flags for which sections have more pages
+        if sectionHasNext.count == 0 {
+            //  since this is the first pass, get indexes of groups
+            //  that we care about (i.e. assignment.count > 0).  Groups
+            //  w/ 0 assignments should not be shown.  All groups are returned
+            //  with paging, so we need to keep indexes of groups we care about
+            response.groups.forEach {
+                if $0.assignments.count > 0 {
+                    groupIDsWithAssignments[$0.id.value] = $0.id.value
+                }
+            }
+            sectionHasNext = Array(repeating: false, count: groupIDsWithAssignments.count)
+        }
+
+        groups = response.groups.filter { groupIDsWithAssignments[ $0.id.value ] != nil }
+        gradingPeriods = response.gradingPeriods
+
+        //  append new assignments to existing arrays
+        for ( index, g ) in groups.enumerated() {
+            let groupID = g.id.value
+            var existing = assignments[groupID] ?? []
+            //  if it's the first run or hasNextPage == true, then append assignments
+            if existing.count == 0 || sectionHasNext[index] == true {
+                existing += g.assignments
+                assignments[groupID] = existing
+            }
+
+            //  handle paging cursor
+            if let pageInfo = g.pageInfo {
+                sectionHasNext[index] = pageInfo.hasNextPage
+            }
+        }
+
+        //  cursor should be the same for all assignmentConnections
+        pagingCursor = groups.compactMap { $0.pageInfo?.endCursor }.first
+
+        if !(shouldFilter) {
+            shouldFilter = true
+            if let currentPeriod = gradingPeriods.current { selectedGradingPeriod = currentPeriod }
+        }
+
+        performUIUpdate { [weak self] in
+            self?.tableView.reloadData()
+            self?.showSpinner(show: false)
+            self?.updateLabels()
+            self?.updateFilterButton()
+            self?.selectFirstCellOnIpad()
+            self?.loading = false
+        }
     }
 
-    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return presenter?.assignments.count ?? 0
+    func resetAPIRequestState() {
+        assignments = [:]
+        pagingCursor = nil
+        sectionHasNext = []
+        fetchedRequests = [:]
+        groupIDsWithAssignments = [:]
     }
 
-    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeue(UITableViewCell.self, for: indexPath)
-        cell.textLabel?.text = presenter?.assignments[indexPath]?.name
-        cell.imageView?.image = .icon(.assignment, .line)
-        cell.imageView?.tintColor = color
+    func selectFirstCellOnIpad() {
+        let ip = IndexPath(row: 0, section: 0)
+        if assignment(for: ip) != nil, splitViewController != nil, !isInSplitViewDetail {
```

Comment:
Hmm this still pushed the first row on iPad when this view is in the detail. narmstrong > s > CS 1400 > Pages > Links to Stuff > Assignments: Link. `isInSplitViewDetail` must be incorrectly returning false. I wonder why. 🤔